File: src/me/krickl/memebotj/InternalCommands/AdminCommands/SaveCommand.scala
package me.krickl.memebotj.InternalCommands.AdminCommands

import me.krickl.memebotj._
import me.krickl.memebotj.Utility.CommandPower

//remove if not needed
import scala.collection.JavaConversions._

class SaveCommand(channel: String, command: String, dbprefix: String) extends CommandHandler(channel,
	command, dbprefix) {

	this.setNeededCommandPower(CommandPower.adminAbsolute)

	this.setHelptext(Memebot.formatText("SAVE_SYNTAX", channelOriginHandler, null, this, true, Array()))

	override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
		channelHandler.sendMessage(Memebot.formatText(channelHandler.localisation.localisedStringFor("SAVE"), channelHandler, sender, this, false, Array()), this.getChannelOrigin)
		for (ch <- Memebot.joinedChannels) {
			ch.writeDBChannelData()
		}
	}
}

File: src/main/java/me/krickl/memebotj/Commands/Internal/SaveCommand.java
package me.krickl.memebotj.Commands.Internal;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.Commands.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.UserHandler;
import me.krickl.memebotj.Utility.CommandPower;

/**
 * This file is part of memebotj.
 * Created by unlink on 11/04/16.
 */
public class SaveCommand extends CommandHandler {
    public SaveCommand(ChannelHandler channelHandler, String commandName, String dbprefix) {
        super(channelHandler, commandName, dbprefix);
    }

    @Override
    public void overrideDB() {
        this.setNeededCommandPower(CommandPower.adminAbsolute);
        this.setHelptext(Memebot.formatText("SAVE_SYNTAX", getChannelHandler(), null, this, true, new String[]{}, ""));
    }

    @Override
    public void commandScript(UserHandler sender, String[] data) {
        getChannelHandler().sendMessage(
                Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("SAVE"), getChannelHandler(),
                        sender, this, false, new String[]{}, ""), getChannelHandler().getChannel(), sender, isWhisper());
        Memebot.saveAll();
    }
}
